spi/src/main/java/org/mapstruct/spi/AccessorNamingStrategy.java

```diff
@@ -0,0 +1,134 @@
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.mapstruct.spi;
+
+/**
+ * A service provider interface for the mapping between method names and properties.
+ *
+ * @author Christian Schuster
+ */
+public interface AccessorNamingStrategy {
+
+    /**
+     * Set the default {@link AccessorNamingStrategy} implementation. Custom implementations may use it to
+     * keep parts of the default behavior. This method is invoked before any other method of the interface.
+     * <p>
+     * Default implementation: Does nothing.
+     *
+     * @param defaultAccessorNamingStrategy The default implementation.
+     */
+    void setDefaultAccessorNamingStrategy(AccessorNamingStrategy defaultAccessorNamingStrategy);
+
+    /**
+     * Determine if a method name defines a getter with a non-boolean return type.
+     * <p>
+     * Default implementation: Method name starts with "get" and is longer than 3 characters.
+     *
+     * @param methodName The method name.
+     * @return {@code true} if the method name can be a non-boolean getter, {@code false} otherwise.
+     */
+    boolean isNonBooleanGetterName(String methodName);
+
+    /**
+     * Determine if a method name defines a getter with a boolean return type.
+     * <p>
+     * Default implementation: Method name starts with "is" and is longer than 2 characters.
+     *
+     * @param methodName The method name.
+     * @return {@code true} if the method name can be a boolean getter, {@code false} otherwise.
+     */
+    boolean isBooleanGetterName(String methodName);
+
+    /**
+     * Determine if a method name defines a setter.
+     * <p>
+     * Default implementation: Method name starts with "set" and is longer than 3 characters.
+     *
+     * @param methodName The method name.
+     * @return {@code true} if the method name can be a setter, {@code false} otherwise.
+     */
+    boolean isSetterName(String methodName);
+
+    /**
+     * Determine if a method name defines an adder.
+     * <p>
+     * Default implementation: Method name starts with "add" and is longer than 3 characters.
+     *
+     * @param methodName The method name.
+     * @return {@code true} if the method name can be an adder, {@code false} otherwise.
+     */
+    boolean isAdderName(String methodName);
+
+    /**
+     * Extract the property name from a method name for which {@link #isNonBooleanGetterName(String)} returned
+     * {@code true}.
+     * <p>
+     * Default implementation: Remove first 3 characters ("get") of the method name, and
+     * {@link java.beans.Introspector#decapitalize(String) decapitalize} the result.
+     *
+     * @param methodName The method name, guaranteed to be a non-boolean getter name.
+     * @return The property name corresponding to the method name.
+     */
+    String getPropertyNameForNonBooleanGetterName(String methodName);
```

Comment:
Same here, I feel it'd be enough to have a single method `getPropertyName(String methodName)` instead of the three separate ones. An implementation could do the distinction internally itself.